From 7f0bbb43a553cba69f2b5ab3e0e24316e0e2db3c Mon Sep 17 00:00:00 2001
From: wenlunpeng <wenlunpeng@uniontech.com>
Date: Thu, 9 May 2024 12:43:18 +0800
Subject: [PATCH 188/360] build: add git commitid in changelog

Banner in dmesg can be flushed if too many debug log.
We save it to deb package so even it's not installed
we can still tell which source code it is from.

To check commitid:
If installed, use apt changelog <package-name>. If not,
extract changelog.Debian.gz from package and zcat it.

Signed-off-by: wenlunpeng <wenlunpeng@uniontech.com>
Signed-off-by: WangYuli <wangyuli@uniontech.com>
---
 scripts/package/mkdebian | 1 +
 1 file changed, 1 insertion(+)

diff --git a/scripts/package/mkdebian b/scripts/package/mkdebian
index 3505d5609b8a..5450a8fd2214 100755
--- a/scripts/package/mkdebian
+++ b/scripts/package/mkdebian
@@ -186,6 +186,7 @@ cat <<EOF > debian/changelog
 $sourcename ($packageversion) $distribution; urgency=low
 
   * Custom built Linux kernel.
+  * Git:$(git symbolic-ref --short HEAD 2>/dev/null || true) $(git rev-parse --short HEAD) $(git diff --quiet HEAD || echo "dirty")
 
  -- $maintainer  $(date -R)
 EOF
-- 
2.20.1

